src/Fetch.js

```diff
@@ -314,21 +314,22 @@ function fetchXHR(fetch, onsuccess, onerror, onprogress, onreadystatechange) {
       xhr.setRequestHeader(keyStr, valueStr);
     }
   }
-  Fetch.xhrs.push(xhr);
-  var id = Fetch.xhrs.length;
-  HEAPU32[fetch + {{{ C_STRUCTS.emscripten_fetch_t.id }}} >> 2] = id;
+  var id = HEAPU32[fetch + {{{ C_STRUCTS.emscripten_fetch_t.id }}} >> 2];
```

Comment:
We can/should probably use `var id = {{{ makeGetValue('fetch', C_STRUCTS.emscripten_fetch_t.id, 'i32' }}}`; here.. but since the existing code uses direct heap access I'll leave it up to you if you want to change that here.